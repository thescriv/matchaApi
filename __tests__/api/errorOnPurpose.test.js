const { startApi, stopApi } = require('../../src/api')
const { apiClient } = require('../apiClient')

describe('errorOnPurpose API', () => {
  let client = null

  beforeAll(async () => {
    await startApi(3001)

    client = new apiClient(3001)
  })

  afterAll(async () => {
    await stopApi()
  })

  describe('GET /error/classic', () => {
    test('do get an error', async () => {
      let error

      try {
        await client.classicErrorOnPurpose()
      } catch (err) {
        error = err.response
      }
      const { body, status } = error

      expect({ body, status }).toMatchSnapshot()
    })
  })

  describe('GET /error/http', () => {
    test('do get an error', async () => {
      let error

      try {
        await client.httpErrorOnPurpose()
      } catch (err) {
        error = err.response
      }
      const { body, status } = error

      expect({ body, status }).toMatchSnapshot()
    })
  })
})
