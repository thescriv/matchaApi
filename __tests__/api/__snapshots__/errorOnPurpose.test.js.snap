// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorOnPurpose API GET /error/classic do get an error 1`] = `
Object {
  "body": Object {
    "help": "an error occured",
    "message": "an error occured",
  },
  "status": 500,
}
`;

exports[`errorOnPurpose API GET /error/http do get an error 1`] = `
Object {
  "body": Object {
    "help": "An error occured",
    "message": "An error occured",
  },
  "status": 400,
}
`;
